test/jdk/tools/sincechecker/SinceChecker.java

```diff
@@ -0,0 +1,828 @@
+ * questions.
+ */
+
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.lang.Runtime.Version;
+import java.net.URI;
+import java.nio.file.*;
+import java.util.*;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+import javax.lang.model.element.*;
+import javax.lang.model.util.ElementFilter;
+import javax.lang.model.util.Elements;
+import javax.lang.model.util.Types;
+import javax.tools.*;
+import javax.tools.JavaFileManager.Location;
+import com.sun.source.tree.*;
+import com.sun.source.util.JavacTask;
+import com.sun.source.util.TreePathScanner;
+import com.sun.source.util.Trees;
+import com.sun.tools.javac.api.JavacTaskImpl;
+import com.sun.tools.javac.code.Flags;
+import com.sun.tools.javac.code.Symbol;
+import com.sun.tools.javac.util.Pair;
+import jtreg.SkippedException;
+
+
+/*
+- This checker checks the values of the `@since` tag found in the documentation comment for an element against the release in which the element first appeared.
+- The source code containing the documentation comments is read from `src.zip` in the release of JDK used to run the test.
+- The releases used to determine the expected value of `@since` tags are taken from the historical data built into `javac`.
+
+The `@since` checker works as a two-step process:
+- In the first step, we process JDKs 9-current, only classfiles, producing a map `<unique-Element-ID`> => `<version(s)-where-it-was-introduced>`.
+    - "version(s)", because we handle versioning of Preview API, so there may be two versions (we use a class with two fields for preview and stable),
+    one when it was introduced as a preview, and one when it went out of preview. More on that below.
+    - For each Element, we compute the unique ID, look into the map, and if there's nothing, record the current version as the originating version.
+    - At the end of this step we have a map of the Real since values
+
+- In the second step, we look at "effective" `@since` tags in the mainline sources, from `src.zip` (if the test run doesn't have it, we throw a `jtreg.SkippedException`)
+    - We only check the specific MODULE whose name was passed as an argument in the test. In that module, we look for unqualified exports and test those packages.
+    -The `@since` checker verifies that for every API element, the real since value and the effective since value are the same, and reports an error if they are not.
+
+Important note : We only check code written since JDK 9 as the releases used to determine the expected value of @since tags
+                 are taken from the historical data built into javac which only goes back that far
+
+note on rules for Real and effective `@since:
+
+Real since value of an API element is computed as the oldest release in which the given API element was introduced. That is:
+- for modules, classes and interfaces, the release in which the element with the given qualified name was introduced
```

Comment:
What about packages?  packages should be in this list as well.

Reply:
Small oversight, will fix